mubench.pipeline/tasks/implementations/requirements_check.py

```diff
@@ -2,27 +2,24 @@
 from typing import List
 
 from requirements import Requirement
-from tasks.project_task import ProjectTask
-from utils.shell import Shell
 
 
-class RequirementsCheck(ProjectTask):
+class RequirementsCheck:
```

Comment:
I'm not sure whether this should be a task at all... it's not doing anything in run, which indicates it isn't. We could turn in into a proper task, if we allow tasks that just return a `continue`/`stop` symbol, i.e., a kind of control task without output. It appears to me this might make sense in other situations as well (e.g., for the info tasks). What do you think?

Reply:
For `stop`, we could use the empty list as we do already. Though we could add some constant like `TaskRunner.stop`  to make this more explicit.
`continue` would be nice, even though we would currently only need that for the info tasks.

Reply:
Do we need stop anywhere, where it is not an exception case (such as "unsuccessful compile" or "all projects filtered")? I'm wondering whether stop is always an exception everything else is a continue, either with new data or with the same data.

What does a function return that has no explicit `return`? Is there a special type/object for that? Is it `None`? Would be nice to implement a "continue" task simply by not returning anything...

Reply:
Upon implementing I noticed that an explicit `stop` is probably not needed, as tasks use exceptions for that. I would probably return `None` if a task has no result, which allows us to omit the return statement in these tasks.

Reply:
This then needs special handling in the runner, right?

    if not result:
      # continue with the same arguments
    else:
      for value in result:
        # continue as before

Reply:
Functions without an explicit `return` return `None`. I'll change the implementation to handle this case like a continue. Whenever we `stop`, at least, we probably want to log a `UserWarning` anyway.

Reply:
Agreed.